### ISSUE TITLE
Real/live time handling for EDS spectrum images

### ISSUE THREAD

**Issue by askorikov** [hyperspy/hyperspy#2534]
Following the discussion in #2362 and #2364, it appears that currently there are inconsistencies in handling real/live time for EDS spectrum images. Specifically, there are several questions to address:
1. Should live and real time values in metadata correspond to time per pixel or per entire spectrum image?
2. I/O plugins for different file formats should provide conforming values.
3. [Metadata structure docs](http://hyperspy.org/hyperspy-doc/current/user_guide/metadata_structure.html) should be updated to clarify this.

**Comment by TomSlater** [comment]
Considering that live time should vary per pixel, the best solution (in my opinion) would be to store a separate array corresponding to live time (and possibly real time) in each pixel. These per pixel values are not accessible from many manufacturers; in this case, perhaps default back to mean live/real time per pixel as standard.

I think there are large inconsistencies in how manufacturers measure/calculate real/live/dead time of their detectors. For example, you can extract per pixel live times from DM4 spectrum images but, depending on how you acquire the data, the live times per pixel can come out orders of magnitude greater than the real time (which is obviously nonsense). It would be great to put together documentation on how each manufacturer calculated their live times as part of updating any I/O plugins.

**Comment by k8macarthur** [comment]
In my experience live_time (the one needed for accurate dose information for zeta and cross section quantification) is typically supplied by the software as per pixel. I agree @TomSlater 's approach of an array would be best practice but currently all the software which offer a live_time only offer one value. Therefore we could modify to code to accept a single value with forward compatibility to accept arrays also.

I'm not sure how useful real_time is. Typically again this is normally one value for the total time of the experiment in my experience. Again issue with consistency between manufacturers exists here also. Is there anyone who uses this value though?

**Comment by ericpre** [comment]
Indeed, live time could be a float or an array. In case of an array, it would more simple to make sure that the array have the same shape as the navigation shape. It may be checking the behaviour of the functions such as `stack`, `split`, etc. functions.

If there are known inconsistency about the `real_time`, then we should remove it from the `metadata` to avoid confusion and spending time on this. It would still be in the `original_metadata`, so it will still be relatively easy to access.

**Comment by sem-geologist** [comment]
Lately I had spend tremendous amount of time in trying to understand how and how-not dead time work in WDS on EPMA (including hardware tear-downs, probing with oscilloscope there and here and other hackish activities). Well, I had not teared down EDS... But I had went through lots of books and other literature to understand what is the difference (and there is some key differences), thus things got quite clear for me. I am going to start including the live_time recalculation for bruker bcf, as I am now sure 100 percent it is very easy to do, but I also can lend a helping hand to other formats. Actually just a moment ago I just solved some interesting issue and tested if that works in #2898 . I am recalling that issue as the problem presented there (vertical stripes on net count results for iron line) extremely well illustrates how it would be completely wrong to address it with the live time as a single value. Those stripe artifacts disappear when dividing results with livetime raster, where every pixel has its own live time. That shows that every pixel needs to have separate live time absolutely, and nothing less.

I would propose that livetime as array would be not stuffed under metadata, but just as attribute under the signal (i.e. `signal.livetime`). metadata could expose (and clearly state) livetime and deadtime as single average values (from my experience it is generally very bad idea stuffing long lists and arrays into `original_metadata`, or `metadata` as that beats performance near to death).

But even better approach is not saving livetime anywhere if it is not in parsed file. Even going more extreme, in case of bruker even if Live time is presented in the file as a raster it is clearly wrong, and correct live time can be generated from 0eV strobe peak. I think better would be to add to metadata if `pixel_live_time_calculable=False` or `True`.  If `True` then io_plugin should pass such function to io reader and function would be linked to method `live_time` (which could be a cached property, or initiated with array if such is present in the given format). 

Anyway, before going with this live time stuff, how it is going to be handled needs to be discussed (@ericpre , @francisco-dlp @TomSlater , @askorikov ). I think I should look through formats and see how many of them embeds live time in the spectra (in particularly 0 keV peak which I is classic)
